### PR TITLE
Add List of Voters on Proposal Page

### PR DESCRIPTION
**Pull request by oliviel**
This PR implements:

- see votes button to see the list of votes.
- A modal to show the list of votes.
- inside the modal, you will see the address, the vote and vp of the user that voted
- link to users VP page.

[issue 135](https://github.com/decentraland/governance/issues/135)

ps: I added this abbreviateNumber function because when the modal is in mobile screens if the number is high will break the text to another line even if the font is smaller in snapshot they do the same if you look at the screenshot.

### CODE REVIEW

**Comment by yemel** [comment]
Please include a description about this PR and link it to the corresponding ticket.

**Comment by yemel** [comment]
### Modify "See Votes" Button
Follow the design attached at the issue: https://github.com/decentraland/governance/issues/135

Do this:
Screen Shot 2021-11-15 at 13 34 58

Instead of this:
Screen Shot 2021-11-15 at 13 35 05


### Replace the address by Display Name if possible

Do this:
Screen Shot 2021-11-15 at 13 40 52

Instead of this:
Screen Shot 2021-11-15 at 13 40 35

**Comment by yemel** [comment]
On Voters list, include link to users VP page:

https://governance.decentraland.org/balance/?address=0x37f00ec67d68c2d3c662339c183d37403c36f28e

File: src/components/Modal/VotesList.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import { Close } from 'decentraland-ui/dist/components/Close/Close'
+import TokenList from 'decentraland-gatsby/dist/utils/dom/TokenList'
+import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+
+import './ProposalModal.css'
+import './VotesList.css'
+
+const SI_SYMBOL = ["", "k", "M", "G", "T", "P", "E"]
+
+function abbreviateNumber(number: number) {
```
Comment (2fd) [inline src/components/Modal/VotesList.tsx:17]: please move this to `src/entities/Votes/utils.ts`

File: src/components/Modal/VotesList.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                <Address value={key} />
+              </div>
+            </Grid.Column>
+            <Grid.Column>
+              <p style={{ marginLeft: '0.5rem' }}>
+                {value.choice === 1 ? l('modal.votes_list.voted_yes') : l('modal.votes_list.voted_no')}
+              </p>
+            </Grid.Column>
+            <Grid.Column>
+              {/* I added this abbreviateNumber function because when the modal is in mobile screens if the number is high
```
Comment (2fd) [inline src/components/Modal/VotesList.tsx:76]: The explanation of what the function does goes in the function
The explanation of why you added it goes in the PR

🙏

File: src/components/Modal/VotesList.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  onClickAccept?: (e: React.MouseEvent<any>) => void
+  votes?: Record<string, Vote> | null,
+}
+
+export function VotesList({onClickAccept, votes, ...props }: VotesListModalProps) {
+  const l = useFormatMessage()
+
+  return <Modal {...props} size="tiny" className={TokenList.join(['ProposalModal', 'VotesList' ,props.className])} closeIcon={<Close />}>
+  <Modal.Content className="ProposalModal__Title">
+    <Header>{Object.entries(votes || {}).length} {Object.entries(votes || {}).length === 1 ? 'Vote' : 'Votes'}</Header>
```
Comment (2fd) [inline src/components/Modal/VotesList.tsx:41]: move this to the `en.json#modal.votes_lis.title` and use `l('modal.votes_lis.title', { votes })`

**Review by 2fd — CHANGES_REQUESTED** [review]

**Comment by oliviel** [comment]
> ### Modify "See Votes" Button
> Follow the design attached at the issue: #135
> 
> Do this: Screen Shot 2021-11-15 at 13 34 58
> 
> Instead of this: Screen Shot 2021-11-15 at 13 35 05
> 
> ### Replace the address by Display Name if possible
> Do this: Screen Shot 2021-11-15 at 13 40 52
> 
> Instead of this: Screen Shot 2021-11-15 at 13 40 35

I change everything here except for the name instead of the address I am facing a problem using the function useAsyncMemo() and profiles.load() is making _rendered more hooks than during the previous render_ I open to suggestions here.

File: src/components/Modal/VotesList.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  onClickAccept?: (e: React.MouseEvent<any>) => void
+  votes?: Record<string, Vote> | null,
+}
+
+export function VotesList({onClickAccept, votes, ...props }: VotesListModalProps) {
+  const l = useFormatMessage()
+
+  return <Modal {...props} size="tiny" className={TokenList.join(['ProposalModal', 'VotesList' ,props.className])} closeIcon={<Close />}>
+  <Modal.Content className="ProposalModal__Title">
+    <Header>{Object.entries(votes || {}).length} {Object.entries(votes || {}).length === 1 ? l('modal.votes_list.vote') : l('modal.votes_list.votes')}</Header>
```
Comment (2fd) [inline src/components/Modal/VotesList.tsx:28]: jajaj, I was specting to use a plural definition of intl:

```json5
    // en.json
    {
        "votes": "{votes} {votes, plural, one {Vote} other {Votes}}",
    }
```

and in code:
```jsx
    <Header>{l('modal.votes_list.votes', { votes: Object.keys(votes || {}).length })}</Header>
```

as ref:
https://github.com/decentraland/governance/blob/master/src/intl/en.json#L4

**Review by 2fd — CHANGES_REQUESTED** [review]

**Review by 2fd — APPROVED** [review]

File: src/components/Modal/VotesList.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        </Grid.Column>
+        <Grid.Column>
+          <div className="VotesList_Header">{l('modal.votes_list.vp')}</div>
+        </Grid.Column>
+      </Grid.Row>
+    </Grid>
+  </div>
+  <div className="VotesList_Container_Items">
+    <Grid columns='equal' className="VotesList_Divider">
+      {Object.entries(votes || {}).sort((a, b) => b[1].vp - a[1].vp).map(vote => {
```
Comment (2fd) [inline src/components/Modal/VotesList.tsx:47]: to load profiles names:
- save the resulte of `Object.entries(votes || {}).sort((a, b) => b[1].vp - a[1].vp)` using a `useMemo`
- create a new component `<VoteListItem key={key} address={key} value={value} />`
- use the `useAsynMemo` hook inside each `VoteListItem` to load the avatar info and show the name

**Review by 2fd — CHANGES_REQUESTED** [review]

File: src/components/Modal/VotesList.tsx (hunk 5 above)
Comment (2fd) [reply src/components/Modal/VotesList.tsx:47]: this could help @oliviel

File: src/components/Section/ProposalResultSection.tsx [hunk 6]
```diff
@@ -51,8 +53,11 @@ export default React.memo(function ProposalResultSection({ proposal, loading, di
   ])}>
     <div className="DetailsSection__Content">
       <Loader active={loading} />
-      <div>
+      <div className="DetailsSection__Flex_Header_Labels">
         <Header sub>{l('page.proposal_detail.result_label')}</Header>
+        {Object.keys(votes || {}).length > 0 && <Anchor style={{ marginBottom: '5px', textTransform: 'uppercase' }} onClick={onOpenVotesList} >
```
Comment (2fd) [inline src/components/Section/ProposalResultSection.tsx:58]: - cache the result of `Object.keys(votes || {}).length` using a `useMemo`
- use `font-size: 13px`
- move the css into the css file

File: src/components/Modal/VotesList.css [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+  margin: 0 auto;
+  width: 90%;
+  margin-bottom: 20px;
+}
+
+.VotesList .VotesList_Container_Items {
+  margin: 0 auto;
+  width: 90%;
+  margin-bottom: 40px;
+  max-height: calc(70vh);
```
Comment (2fd) [inline src/components/Modal/VotesList.css:11]:
```suggestion
  max-height: calc(70vh);
```
```suggestion
  height: 50vh;
```

File: src/components/Modal/VotesList.css [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  padding: 20px 5px;
+  overflow-y: scroll;
```
Comment (2fd) [inline src/components/Modal/VotesList.css:13]:
```suggestion
  overflow-y: scroll;
```
```suggestion
  overflow-y: auto;
```

**Review by 2fd — CHANGES_REQUESTED** [review]

**Comment by 2fd** [comment]
Instead of hiding  `see votes` y prefer to show it as disable

Screen Shot 2021-11-18 at 11 07 31

@yemel WDYT?

**Comment by yemel** [comment]
Generally, I like better the option of disabling the action button when data is not available.
But since the UI already has a lot of info I think it's ok to hide it.

**Review by 2fd — APPROVED** [review]

**Comment by 2fd** [comment]
Awesome work! Thanks 😄

**Pull request merged**